### PR TITLE
Adjust Wallet Interface

### PR DESCRIPTION
**Pull request by dternyak**
Closes https://github.com/MyCryptoHQ/MyCrypto/issues/1153

To test:

1. test generate keystore works (from https://github.com/MyCryptoHQ/MyCrypto/pull/1154/files#diff-da809197b595406e304acd50516913eaR62)
2. test web3 unlock works (https://github.com/MyCryptoHQ/MyCrypto/pull/1154/files#diff-6e02304097be0810552c805f4537b71bR18)
3. Test signing works for all formats (https://github.com/MyCryptoHQ/MyCrypto/pull/1154/files#diff-14d64f27973fa652c812d4bf57981601R28)
4. Test payment requests work (https://github.com/MyCryptoHQ/MyCrypto/pull/1154/files#diff-3b47f7a518452473a0a206bddecfd7b9R144)
5. Test creating a signed message on all wallet formats (https://github.com/MyCryptoHQ/MyCrypto/pull/1154/files#diff-4922ccfe05f56e9512da9a9c6571d78dR52

### CODE REVIEW

File: common/containers/Tabs/SignAndVerifyMessage/components/SignMessage/SignButton.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export default class SignMessageButton extends React.Component<Props, {}> {
 
     try {
       const signedMessage: ISignedMessage = {
-        address: await wallet.getAddressString(),
+        address: wallet.getAddressString(),
```
Comment (wbobeirne) [inline common/containers/Tabs/SignAndVerifyMessage/components/SignMessage/SignButton.tsx]: This function no longer needs to be async.

File: common/utils/localStorage.ts [hunk 2]
```diff
@@ -40,16 +40,16 @@ export async function saveWalletConfig(
   wallet: IWallet,
   state: Partial<WalletConfig>
 ): Promise<WalletConfig> {
-  const oldState = await loadWalletConfig(wallet);
+  const oldState = loadWalletConfig(wallet);
   const newState = { ...oldState, ...state };
-  const key = await getWalletConfigKey(wallet);
+  const key = getWalletConfigKey(wallet);
```
Comment (wbobeirne) [inline common/utils/localStorage.ts]: this function no longer needs to be async

File: common/containers/Tabs/SignAndVerifyMessage/components/SignMessage/SignButton.tsx (hunk 1 above)
Comment (dternyak) [reply common/containers/Tabs/SignAndVerifyMessage/components/SignMessage/SignButton.tsx]: It does due to `sig: await wallet.signMessage(message),` unless I am mistaken

File: common/utils/localStorage.ts (hunk 2 above)
Comment (dternyak) [reply common/utils/localStorage.ts]: thanks, fixed.

**Comment by dternyak** [comment]
@wbobeirne Perhaps https://github.com/MyCryptoHQ/MyCrypto/issues/1137 could be a connected component that retrieves address, checksums, and then renders? Either way I suggest we handle that in subsequent PR.

File: common/containers/Tabs/SignAndVerifyMessage/components/SignMessage/SignButton.tsx (hunk 1 above)
Comment (wbobeirne) [reply common/containers/Tabs/SignAndVerifyMessage/components/SignMessage/SignButton.tsx]: Well butter my biscuits, you're right. My bad.

**Pull request merged**